### PR TITLE
Implemented a Build, Test, and Lint CI Pipeline

### LINKED ISSUE #39: [US] FrontEnd Build Code with CI Pipeline
# Description
As developer, I want to have a CI Pipeline to make a build after every commit.

# Acceptance criteria
CI Pipeline builds code after code is pushed to GitHub.

# Possible way to test
GitHub Actions Tab Entries.

### LINKED ISSUE #41: [US] FrontEnd Run Unit Tests with CI Pipeline
# Description
As developer, I want to have run all unit tests on every push to ensure that the code is working.

# Acceptance criteria
All unit tests run when the code is pushed.

# Possible way to test
Write unit tests and push the code.

### LINKED ISSUE #42: [US] FrontEnd Check Code Style and Syntax with CI Pipeline
# Description
As  developer, I want to check the code style and syntax on every push to make sure that the code is well formatted and correct.

# Acceptance criteria
CI pipeline checks syntax and style on every push and gives feedback.

# Possible way to test
Push bad formatted code into GitHub and check if the CI gives feedback.

### LINKED ISSUE #43: [US] Different CI Pipelines for Push and Pull_Request
# Description
As developer, I want to have different ci pipelines to have less checks for regular pushes and more checks for pull requests.

# Acceptance criteria
Pull request pipeline executes a build, test and lint. Push pipeline executes only build and test.

# Possible way to test
Make a normal push and make a pull request.

### LINKED ISSUE #44: [US] Format Code on the FrontEnd before it is pushed
# Description
As developer, I want to have a well formatted code to push always well structured code on github.

# Acceptance criteria
Code gets formatted before it is pushed to github.

# Possible way to test
Write not well formatted code and push it to github.

### LINKED ISSUE #45: [US] Implement Review Suggestions on CI Pipelines
# Description
As developer, I want to have ci pipeline names that are clear to understand the purpose of the ci pipeline fast.

# Acceptance criteria
CI Pipelines have clear names.

# Possible way to test
Review from another member.

### PR DESCRIPTION
**Pull request by Antonius-MS**
# 💻 What
Implemented a CI pipeline with build, test and lint for FrontEnd for on push and on pull_request. Additionally implemented a formatter for pre-commit.

# ❓ Why
To check if the pushed code is working.

# 📘 How
Github actions and workflows.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,10 +6,14 @@
...
     "watch": "ng build --watch --configuration development",
-    "test": "ng test"
+    "test": "ng test --browsers=ChromeHeadless --watch=false --code-coverage",
+    "lint": "node ./node_modules/eslint/bin/eslint src --ext .js,.jsx,.ts,.tsx",
+    "precommit": "pretty-quick --staged",
+    "prepare": "husky install .husky"
   },
   "private": true,
   "dependencies": {
+    "@angular-eslint/schematics": "^15.2.1",
```
Comment (dev-Fabi) [inline package.json]: Hmm, is this not also just a dev dependency?

File: package.json [hunk 2]
```diff
@@ -24,15 +28,27 @@
   },
   "devDependencies": {
     "@angular-devkit/build-angular": "^15.2.4",
+    "@angular-eslint/eslint-plugin": "^15.2.1",
     "@angular/cli": "~15.2.4",
     "@angular/compiler-cli": "^15.2.0",
     "@types/jasmine": "~4.3.0",
+    "@types/prettier": "2.7.2",
```
Comment (dev-Fabi) [inline package.json]: I guess the type definitions for prettier are not really needed?

File: .github/workflows/ci-pullrequest-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: Web Frontend CI Pipeline on PULL REQUEST
```
Comment (dev-Fabi) [inline .github/workflows/ci-pullrequest-pipeline.yml]:
```suggestion
name: Check Style
```

Give a meaning full name to the workflows so it is clear what they do. Same for the filename :)

File: .github/workflows/ci-pullrequest-pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+      - name: Build
+        run: |
+          npm run build
+      - name: Lint
+        run: |
+          npm run lint
+      - name: Test
+        run: |
+          npm run test
```
Comment (dev-Fabi) [inline .github/workflows/ci-pullrequest-pipeline.yml]:
```suggestion
      - name: Lint
        run:  npm run lint
```

The other workflow already executes the tests. Not needed to run them twice.

File: .github/workflows/ci-push-pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+name: Web Frontend CI Pipeline on PUSH
```
Comment (dev-Fabi) [inline .github/workflows/ci-push-pipeline.yml]:
```suggestion
name: Test
```

File: .github/workflows/ci-push-pipeline.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Install ⚙️
+        run: npm ci
+
+      - name: Build
+        run: |
+          npm run build
+
+      - name: Test
+        run: |
+          npm run test
```
Comment (dev-Fabi) [inline .github/workflows/ci-push-pipeline.yml]:
```suggestion
      - name: Build
        run: npm run build

      - name: Test
        run: npm run test
```

**Review by dev-Fabi — APPROVED** [review]
Nice 🚀

File: package.json (hunk 1 above)
Comment (Antonius-MS) [reply package.json]: True, done

File: package.json (hunk 2 above)
Comment (Antonius-MS) [reply package.json]: No, done

File: .github/workflows/ci-pullrequest-pipeline.yml (hunk 3 above)
Comment (Antonius-MS) [reply .github/workflows/ci-pullrequest-pipeline.yml]: Done

File: .github/workflows/ci-pullrequest-pipeline.yml (hunk 4 above)
Comment (Antonius-MS) [reply .github/workflows/ci-pullrequest-pipeline.yml]: True, done

**Review by dev-Fabi — APPROVED** [review]

**Pull request merged**